autoPyTorch/pipeline/components/training/trainer/base_trainer_choice.py

```diff
@@ -506,3 +599,31 @@ def __str__(self) -> str:
         """ Allow a nice understanding of what components where used """
         string = str(self.run_summary)
         return string
+
+    def _get_search_space_updates(self, prefix: Optional[str] = None) -> Dict[str, HyperparameterSearchSpace]:
+        """Get the search space updates with the given prefix
+
+        Keyword Arguments:
+            prefix {str} -- Only return search space updates with given prefix (default: {None})
+
+        Returns:
+            dict -- Mapping of search space updates. Keys don't contain the prefix.
+        """
+        updates = super()._get_search_space_updates(prefix=prefix)
+
+        result: Dict[str, HyperparameterSearchSpace] = dict()
+
+        # iterate over all search space updates of this node and filter the ones out, that have the given prefix
+        for key in updates.keys():
+            if Lookahead.__name__ in key:
+                # need to also remove lookahead from the hyperparameter name
+                new_update = HyperparameterSearchSpace(
+                    updates[key].hyperparameter.replace('{}:'.format(Lookahead.__name__), ''),
+                    value_range=updates[key].value_range,
+                    default_value=updates[key].default_value,
+                    log=updates[key].log
+                )
+                result[key.replace('{}:'.format(Lookahead.__name__), '')] = new_update
+            else:
+                result[key] = updates[key]
```

Comment:
Is this `updates` dict a tmp variable?
If so, it does not matter, but otherwise, we must copy.